nBTC/sources/redeem_request.move

```diff
@@ -61,13 +61,16 @@ public struct RedeemRequest has store {
     created_at: u64,
 }
 
-//TODO: Add logic to extract data from redeem inputs for:
-/// Event emitted when a proposal for redeem request is selected (solved) and we are ready
-/// for creating MPC signatures.
-public struct SolvedEvent has copy, drop {
+/// Generic event emitted when redeem req changes status.
+/// Status: 0 = Resolving, 1 = Signing, 2 = Signed, 3 = Confirmed
+public struct RedeemReqStatusEvent has copy, drop {
     id: u64,
-    utxo_ids: vector<u64>,
-    dwallet_ids: vector<ID>,
+    status: u8,
+    utxo_ids: Option<vector<u64>>, // when moving to Signing
+    dwallet_ids: Option<vector<ID>>, // when moving to Signing
+    input_idx: Option<u32>, // during signature validation
+    is_fully_signed: Option<bool>, // during signature validation
+    btc_tx_id: Option<vector<u8>>, // when Confirmed
```

Comment:
Actually, this is not a good idea. Sorry - my mistake. I thought we have more events that differen only by status.